SofaKernel/framework/sofa/simulation/Node.h

```diff
@@ -538,39 +561,48 @@ class SOFA_SIMULATION_CORE_API Node : public sofa::core::objectmodel::BaseNode,
     /// return the smallest common parent between this and node2 (returns NULL if separated sub-graphes)
     virtual Node* findCommonParent( simulation::Node* node2 ) = 0;
 
-	/// override context setSleeping to add notification.
-	void setSleeping(bool /*val*/) override;
+    /// override context setSleeping to add notification.
+    void setSleeping(bool /*val*/) override;
+
 
 protected:
     bool debug_;
     bool initialized;
 
-    virtual void doAddObject(sofa::core::objectmodel::BaseObject::SPtr obj);
-    virtual void doRemoveObject(sofa::core::objectmodel::BaseObject::SPtr obj);
-
+    virtual bool doAddObject(sofa::core::objectmodel::BaseObject::SPtr obj);
+    virtual bool doRemoveObject(sofa::core::objectmodel::BaseObject::SPtr obj);
+    virtual void doMoveObject(sofa::core::objectmodel::BaseObject::SPtr sobj, Node* prev_parent);
 
     std::stack<Visitor*> actionStack;
+private:    
+    virtual void notifyBeginAddChild(Node::SPtr parent, Node::SPtr child);
+    virtual void notifyBeginRemoveChild(Node::SPtr parent, Node::SPtr child);
```

Comment:
`notify*Begin`?

Reply:
I just realized I went for the other way around, `notify{Add,Remove,Move}XXX{Begin,End}` instead of `notify{Begin,End}{Add,Remove,Move}XXX`. I guess as long as it's uniform throughout the code base, it should be alright.. If that's OK for you